### PR TITLE
datasource network client and CI pipeline

### PR DESCRIPTION
**Pull request by frombrest**

### CODE REVIEW

File: shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt [hunk 1]
```diff
@@ -7,12 +7,16 @@ class DataDeserializationTest {
...
-        assertTrue(data.linkTypes.size == 3)
-        assertTrue(data.publishers.size == 3)
+
+        assertTrue {
+            data.books.size == 3 &&
+            data.people.size == 3 &&
+            data.tags.size == 3 &&
+            data.linkTypes.size == 3 &&
+            data.publishers.size == 3
+        }
```
Comment (nbeloglazov) [inline shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt:15]: Why this change? Having individual asserts is easier for debugging.

File: shared/src/commonTest/kotlin/by/audiobooks/mob/network/SiteApiTest.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+        val data = SiteApi().downloadData()
+
+        assertTrue {
+            data.books.isNotEmpty() &&
+            data.tags.isNotEmpty() &&
+            data.people.isNotEmpty() &&
+            data.linkTypes.isNotEmpty() &&
+            data.publishers.isNotEmpty()
+        }
```
Comment (nbeloglazov) [inline shared/src/commonTest/kotlin/by/audiobooks/mob/network/SiteApiTest.kt:16]: Same here.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: Build
```
Comment (nbeloglazov) [inline .github/workflows/build.yml:1]: This workflow is for building the project. Do you have another for running tests? I see that you have few tests.

Comment (frombrest) [reply .github/workflows/build.yml:1]: Build includes running tests for both platforms

File: shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt (hunk 1 above)
Comment (frombrest) [reply shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt:15]: Actually it does not any matter, and any single assertion will be more that enough since deserialization will fail if json payload will have an inexactness. 
I can delete redundant verifications if you like.

File: shared/src/commonTest/kotlin/by/audiobooks/mob/network/SiteApiTest.kt (hunk 2 above)
Comment (frombrest) [reply shared/src/commonTest/kotlin/by/audiobooks/mob/network/SiteApiTest.kt:16]: Answered in previous comment ⬆️

File: .github/workflows/build.yml (hunk 3 above)
Comment (frombrest) [reply .github/workflows/build.yml:1]: CI runs on ephemeral node, so it does not store resulted artifacts... Later I guess we will add separated manual pipeline to release.

Comment (nbeloglazov) [reply .github/workflows/build.yml:1]: Just to confirm `gradle build` both compiles and runs tests? That's a bit confusing.

File: shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt (hunk 1 above)
Comment (nbeloglazov) [reply shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt:15]: I see. If the the main goal of the test to verify that it correctly parses JSON then doing all those checks is reduntant indeed. Probably keep just 1 a sanity check.

**Review by nbeloglazov — APPROVED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (frombrest) [reply .github/workflows/build.yml:1]: Right... Honestly I am not an expert with gradle (working with maven mostly) so empirically I defined that `test` goal does not run test for iOS platform for some reason.

File: shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt (hunk 1 above)
Comment (frombrest) [reply shared/src/commonTest/kotlin/by/audiobooks/mob/model/source/DataDeserializationTest.kt:15]: fixed
